packages/stark-ui/src/modules/slider/components/slider.component.ts

```diff
@@ -0,0 +1,178 @@
+import {
+	AfterViewInit,
+	Component,
+	ElementRef,
+	EventEmitter,
+	HostBinding,
+	Inject,
+	Input,
+	OnChanges,
+	OnInit,
+	Output,
+	SimpleChanges,
+	ViewEncapsulation
+} from "@angular/core";
+
+import * as noUiSliderLibrary from "nouislider";
+
+import { STARK_LOGGING_SERVICE, STARK_ROUTING_SERVICE, StarkLoggingService, StarkRoutingService } from "@nationalbankbelgium/stark-core";
+
+import { StarkDOMUtil } from "../../../util/dom.util";
+import { StarkSliderConfig } from "./slider-config.intf";
+
+/**
+ * @ignore
+ */
+const _isEqual: Function = require("lodash/isEqual");
+
+/**
+ * Name of the component
+ */
+const componentName: string = "stark-slider";
+
+/**
+ * Component to display a slider with one or more handles
+ */
+@Component({
+	selector: "stark-slider",
+	templateUrl: "./slider.component.html",
+	encapsulation: ViewEncapsulation.None
+})
+export class StarkSliderComponent implements AfterViewInit, OnChanges, OnInit {
+	/**
+	 * Adds class="stark-slider" attribute on the host component
+	 */
+	@HostBinding("class") public class: string = componentName;
+
+	/**
+	 * Whether the slider is disabled. Default: false
+	 */
+	@Input() public isDisabled?: boolean;
+
+	/**
+	 * Configuration object for the slider instance to be created.
+	 */
+	@Input() public sliderConfig: StarkSliderConfig;
+
+	/**
+	 * HTML "id" attribute of the element.
+	 */
+	@Input() public sliderId: string;
+
+	/**
+	 * Array of numeric values to be set to the slider.
+	 * For simple sliders with just one handle, the array should contain only one value.
+	 * For range sliders, the array should contain two or more values.
+	 */
+	@Input() public values: number[];
+
+	/**
+	 * Event to be emitted when the slider's value(s) change.
+	 */
+	@Output() public changed?: EventEmitter<number[]> = new EventEmitter();
+
+	/**
+	 * Stores the latest value, to be able to see if values have been changed
+	 */
+	public latestUnencodedValues: number[];
+
+	/**
+	 * set to true if the slider is in horizontal mode
+	 */
+	public isHorizontal: boolean;
+
+	/**
+	 * a reference to the noUiSlider component inside the noUiSlider library
+	 */
+	public slider: noUiSliderLibrary.noUiSlider;
+
+	/**
+	 * to store the ElementRef
+	 */
+	public elementRef: ElementRef;
```

Comment:
You can remove this property from here and define it directlyl in the constructor like this:

``` typescript
public constructor(public elementRef: ElementRef) {
}
```